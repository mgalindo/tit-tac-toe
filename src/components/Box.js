import React from "react";
import PropTypes from "prop-types";
import {
  Text,
  View,
  StyleSheet,
  Dimensions,
  TouchableOpacity
} from "react-native";

const DeviceWidth = Dimensions.get("window").width;

const Box = props => {
  const { value, index, onPlayerMove, curPlayer } = props;
  const disabled = value.trim() !== "";
  let valueStyle = value.trim() === "" ? {} : styles["box" + value];

  return (
    <View style={[styles.box, valueStyle]}>
      <TouchableOpacity
        style={styles.button}
        onPress={() => {
          if (!disabled) {
            onPlayerMove(index, curPlayer);
          }
        }}
        disabled={disabled}
        testID="box-button"
      >
        <Text style={styles.boxText} testID="box-text">
          {value}
        </Text>
      </TouchableOpacity>
    </View>
  );
};

Box.propTypes = {
  value: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  curPlayer: PropTypes.string.isRequired,
  onPlayerMove: PropTypes.func.isRequired
};

const styles = StyleSheet.create({
  box: {
    width: DeviceWidth * 0.2,
    height: DeviceWidth * 0.2,
    marginBottom: 5,
    marginLeft: 6,
    backgroundColor: "#4A8305",
    justifyContent: "center"
  },
  boxX: {
    backgroundColor: "grey"
  },
  boxO: {
    backgroundColor: "red"
  },
  button: {
    alignItems: "center"
  },
  boxText: {
    fontSize: 40,
    fontWeight: "bold"
  }
});

export default Box;
