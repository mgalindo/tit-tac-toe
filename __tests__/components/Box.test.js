import React from "react";

import { render } from "react-native-testing-library";

import Box from "../../src/components/Box";
import { italic } from "ansi-colors";

describe("Box component", () => {
  let props;
  let onPlayerMove;

  beforeEach(() => {
    const onPlayerMove = jest.fn();
    props = { value: "", index: 0, curPlayer: "X", onPlayerMove: onPlayerMove };
  });

  test("should render without value", () => {
    expect.assertions(1);
    const { toJSON } = render(<Box {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  test("should render with X value", () => {
    expect.assertions(1);
    props.value = 'X';
    const { toJSON } = render(<Box {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  test("should render with O value", () => {
    expect.assertions(1);
    props.value = 'O';
    const { toJSON } = render(<Box {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });  
  
});
