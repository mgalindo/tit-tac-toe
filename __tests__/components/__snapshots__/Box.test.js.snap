// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component should render with O value 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#4A8305",
        "height": 150,
        "justifyContent": "center",
        "marginBottom": 5,
        "marginLeft": 6,
        "width": 150,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 40,
          "fontWeight": "bold",
        }
      }
    >
      O
    </Text>
  </View>
</View>
`;

exports[`Box component should render with X value 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#4A8305",
        "height": 150,
        "justifyContent": "center",
        "marginBottom": 5,
        "marginLeft": 6,
        "width": 150,
      },
      Object {
        "backgroundColor": "grey",
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 40,
          "fontWeight": "bold",
        }
      }
    >
      X
    </Text>
  </View>
</View>
`;

exports[`Box component should render without value 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#4A8305",
        "height": 150,
        "justifyContent": "center",
        "marginBottom": 5,
        "marginLeft": 6,
        "width": 150,
      },
      Object {},
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 40,
          "fontWeight": "bold",
        }
      }
    >
      
    </Text>
  </View>
</View>
`;
